tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -68,15 +66,68 @@ make_single_value_arrow(Oid pgtype, Datum datum, bool isnull)
 		FOR_TYPE(TIMESTAMPOID, Timestamp, DatumGetTimestamp);
 		FOR_TYPE(DATEOID, DateADT, DatumGetDateADT);
 		default:
-			elog(ERROR, "unexpected column type '%s'", format_type_be(pgtype));
+			elog(ERROR, "unexpected column type '%s'", format_type_be(arithmetic_type));
 			pg_unreachable();
 	}
 
-	arrow_set_row_validity(&with_buffers->nulls_buffer, 0, true);
+	return arrow;
+}
+
+static ArrowArray *
+make_single_value_arrow_text(Datum datum, bool isnull)
+{
+	struct ArrowWithBuffers
+	{
+		ArrowArray arrow;
+		uint64 arrow_buffers_array_storage[3];
+		uint64 validity_buffer[1];
+		uint32 offsets_buffer[2];
+		uint64 values_buffer[8 /* 64-byte padding as required by Arrow. */];
```

Comment:
Nit: I think it is more common to add the comment at the end of the line. Could not find any other comment like this in the PG sources.

```suggestion
		uint64 values_buffer[8]; /* 64-byte padding as required by Arrow. */
```